tiny_dnn/core/kernels/fully_connected_grad_op.h

```diff
@@ -65,32 +67,38 @@ class FullyConnectedGradOp : public core::OpKernel {
     auto params = OpKernel::params_->fully();
 
     // incoming/outcoming data
-    const tensor_t &prev_out = context.input(0);
-    const tensor_t &W        = context.input(1);
-    tensor_t &dW             = context.input_grad(1);
-    tensor_t *db         = params.has_bias_ ? &context.input_grad(2) : nullptr;
-    tensor_t &prev_delta = context.input_grad(0);
-    tensor_t &curr_delta = context.output_grad(0);
-    tensor_t dummy;  // need lvalue for non-const reference
+    const xt::xarray<float_t> prev_out = to_xtensor(context.input(0));
+    const xt::xarray<float_t> W        = to_xtensor(context.input(1));
+    xt::xarray<float_t> dW             = to_xtensor(context.input_grad(1));
+    xt::xarray<float_t> dB             = params.has_bias_
+                               ? to_xtensor(context.input_grad(2))
+                               : xt::xarray<float_t>();  // TODO
+    xt::xarray<float_t> prev_delta = to_xtensor(context.input_grad(0));
+    xt::xarray<float_t> curr_delta = to_xtensor(context.output_grad(0));
+    xt::xarray<float_t> dummy;  // need lvalue for non-const reference
 
     // initialize outputs
-    fill_tensor(prev_delta, float_t{0});
+    prev_delta = xt::zeros<float_t>(prev_delta.shape());
 
     // call the algorithm depending on the selected engine type
 
     const core::backend_t engine = context.engine();
 
```

Comment:
Sure, I just wanted to get a working version first, and then improve it to a standard version which would be expanded to the other layers.